Add docs for taxon constraints (#1708)
obophenotype/cell-ontology: docs/taxon-restrictions.md

> Cell ontology follows certain conventions for adding taxon constraints. Please do go throug the following documents before adding taxon constraints to CL terms. 

Reviewer: ```suggestion
The cell ontology (CL) follows certain conventions regarding taxon constraints. 
Please review the following pages before adding taxon constraints to CL terms. 
```

Explanation of taxon constraints can be found here:
https://oboacademy.github.io/obook/explanation/taxon-constraints-explainer/ 

> A how to guide can be found here:

Reviewer: ```suggestion
A how-to guide can be found here:
```
